firebase-firestore/src/main/java/com/google/firebase/firestore/local/MemoryDocumentOverlayCache.java

```diff
@@ -105,4 +105,32 @@ public void removeOverlaysForBatchId(int batchId) {
 
     return result;
   }
+
+  @Override
+  public Map<DocumentKey, Pair<Integer, Mutation>> getOverlaysWithBatchId(
```

Comment:
Please find a way to remove this duplication - even if we do not add an Overlay class.